proto/substrait/algebra.proto

```diff
@@ -102,7 +102,7 @@ message RelCommon {
 // The scan operator of base data (physical or virtual), including filtering and projection.
 message ReadRel {
   RelCommon common = 1;
-  NamedStruct base_schema = 2;
+  Type.Struct base_schema = 2;
```

Comment:
this would be a breaking change, so I guess we can't do it. I'm just including it here for discussion - if we were to add names to the type itself, then we could replace (through some evolutions) the NamedStruct usages with Type.Struct.